tests/q_functions_tests/basetest_state_action_q_function.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from builtins import *  # NOQA
+from future import standard_library
+standard_library.install_aliases()
+
+import unittest
+
+import chainer
+import chainer.functions as F
+from chainer import testing
+from chainer.testing import attr
+import numpy as np
+
+import chainerrl
+
+
+class _TestSAQFunction(unittest.TestCase):
+
+    def _test_call_given_model(self, model, gpu):
+        # This method only check if a given model can receive random input
+        # data and return output data with the correct interface.
+        batch_size = 7
+        obs = np.random.rand(batch_size, self.n_dim_obs).astype(np.float32)
+        action = np.random.rand(
+            batch_size, self.n_dim_action).astype(np.float32)
+        if gpu >= 0:
+            model.to_gpu(gpu)
+            obs = chainer.cuda.to_gpu(obs)
+            action = chainer.cuda.to_gpu(action)
+        y = model(obs, action)
+        self.assertTrue(isinstance(y, chainer.Variable))
+        self.assertEqual(y.shape, (batch_size, 1))
+        self.assertEqual(chainer.cuda.get_array_module(y),
+                         chainer.cuda.get_array_module(obs))
+
+
+@testing.parameterize(
+    *testing.product({
+        'n_dim_obs': [1, 5],
+        'n_dim_action': [1, 3],
+        'n_hidden_layers': [0, 1, 2],
+        'n_hidden_channels': [1, 2],
+        'nonlinearity': ['relu', 'elu'],
+        'last_wscale': [1, 1e-3],
+    })
+)
+class TestFCSAQFunction(_TestSAQFunction):
```

Comment:
This is a concrete test case, not a base test. It looks the same as `TestFCSAQFunction` in `tests/q_functions_tests/test_state_action_q_function.py`, so I think it is not necessary.